lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1073,6 +1073,12 @@ def cla(self):
             self.zaxis._set_scale(self._sharez.zaxis.get_scale())
         else:
             self.zaxis._set_scale('linear')
+            try:
+                self.set_zlim(0, 1)
+            except TypeError:
+                pass
+            except AttributeError:
```

Comment:
The Axes3D.cla() method is setting up the zaxis before calling the 2D version of cla(). This used to be fine, but the 2D version is what creates `self.callbacks`. The 2D version of cla() used to be quite simple. I am rather surprised at how complicated it has become since I last looked at it.

IIRC, simply moving the 2D's cla() to the top of the call for the 3D cla() would not work, as the axis setting step messes up some of the stuff set later in the method. We can give it a try, but we might need to refactor the 2D cla() into multiple sub-parts that can be called from the 3D cla().